### ISSUE TITLE
example with /app got error 404 using dynamic segment

### ISSUE THREAD

**Issue by chhornponleu** [aralroca/next-translate-plugin#11]
**What version of this package are you using?**
```
{ 
  "dependencies": { 
    "next": "^13.2.1",
    "next-translate": "^2.0.2", 
  },
  "devDependencies": {  
    "next-translate-plugin": "2.0.0", 
  }
}
```
**What operating system, Node.js, and npm version?**
```
Mac M1, Node: v16.19.0, npm: 8.19.3
```

**What happened?**

<img width="278" alt="image" src="https://user-images.githubusercontent.com/14964945/224467240-e3a083c2-fc10-4e26-93dd-5fbb410ee82c.png">

I create dynamic url `[userId]`, and cannot access neither `/use123` or `/user123/profile`. It redirect to error 404 page.

Checkout the repo I extracted from [with-app-directory example](https://github.com/aralroca/next-translate/tree/master/examples/with-app-directory) 

https://github.com/chhornponleu/next-translate-app-dir-demo

**Comment by awinogrodzki** [comment]
I can confirm that the error does also happen with `next-translate` and `next-translate-plugin` both in version `2.0.2`

**Comment by aralroca** [comment]
I realize that without next-translate is not working either. With `i18n` routing and `appDir`:

**next.config.js**
```js
module.exports = {
  experimental: { appDir: true },
  i18n: {
    locales: ['en', 'ca', 'es'],
    defaultLocale: 'en'
  }
}
```

🤔

**Comment by aralroca** [comment]
Related with this issue https://github.com/vercel/next.js/issues/46841

**Comment by aralroca** [comment]
As a workaround meanwhile Next.js is not fixed,  you  can add this the middleware:

```diff
import { NextResponse } from "next/server";
import type { NextRequest } from "next/server";
import i18n from "./i18n";

export function middleware(request: NextRequest) {
  const locale = request.nextUrl.locale || i18n.defaultLocale;
  request.nextUrl.searchParams.set("lang", locale);
+  request.nextUrl.href = request.nextUrl.href.replace(`/${locale}`, "");
  return NextResponse.rewrite(request.nextUrl);
}
```

**Comment by aralroca** [comment]
I have made a PR to support `/[lang]` dynamic routing in an easy way.

You can try it with these versions:

- `next-translate` - version [**`2.3.0-canary.3`**](https://github.com/aralroca/next-translate/releases/tag/2.3.0-canary.3) 
- `next-translate-plugin` - version [**`2.3.0-canary.6`**](https://github.com/aralroca/next-translate-plugin/releases/tag/2.3.0-canary.6)

Things to keep in mind:

- The  pages defined inside `i18n.js` will have to be updated by adding the `/[lang]` in front of them.
- No middleware would be needed, unless you want to control a redirect.
- You have to put all current `/app` pages inside `/app/[lang]`.

I updated the example of next-translate working with `/app/[lang]` and Next.js 13.4:

https://github.com/aralroca/next-translate/blob/2078477ec299d1287cc20d5245e3781554a1ac39/examples/with-app-directory/

**Comment by aralroca** [comment]
both ways now are supported (`?lang=` or `/[lang]`), but the way using `?lang=` needs middleware to do the rewrite and as of Next.js 13.3 it no longer works well, so I've added a second option for people to use until [the issue](https://github.com/vercel/next.js/issues/49883) is resolved by the Next.js team.

**Comment by aralroca** [comment]
Solved in 2.3
